WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java

```diff
@@ -1186,6 +1189,14 @@ public static boolean shouldShowStoriesIntro() {
         return getBoolean(UndeletablePrefKey.SHOULD_SHOW_STORIES_INTRO, true);
     }
 
+    public static void setHasLaunchedGutenbergEditor(boolean hasLaunched) {
+        setBoolean(UndeletablePrefKey.HAS_LAUNCHED_GUTENBERG_EDITOR, hasLaunched);
```

Comment:
nit: "has launched" feels slightly misleading to me because,based on that name, I would expect it to update to `true` as soon as Gutenberg is launched, not when it is closed. With that said, I don't have a clearly better suggestion. The best I've come up with is something like "has used", which isn't great either. I don't view this as a blocker, just raising it in case you might have a better idea.

Reply:
The "has launched" name aligns with my original intent to set the value when the editor launched. However, I was unable to identify an appropriate [`Activity` lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle) that would run _after_ [`GutenbergPropsBuidler` was initialized](https://github.com/wordpress-mobile/WordPress-Android/blob/04e523e0505945507852b57db0f57d752bcc6b1a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L2260-L2298) for the first time. Both `onCreate` and `onStart` appeared to run prior to the `GutenbergPropsBuilder`. 

The iOS implementation follows my original intent where [`gutenbergDidMount ` runs](https://github.com/wordpress-mobile/WordPress-iOS/blob/3e47fc112bcf8eb31192adff0847ac875d73310f/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L831-L841) after the initial call of [`gutenbergCapabilities`](https://github.com/wordpress-mobile/WordPress-iOS/blob/3e47fc112bcf8eb31192adff0847ac875d73310f/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L1048-L1060). So, the flag is truly set when the editor "launches" rather than when it closes. 

Are you aware of any lifecycles we might be able to hook into to make the "has launched" name continue to make sense? If not, do you believe it is better to retain the "has launched" name to match iOS or rename the Android name to more accurately represent the point at which it's set? 

I am also not dedicated to the "has launched" name on iOS. I think `hasUsedGutenbergEditor` could work for both Android and iOS, if you feel that is clearer.

Reply:
My initial thought was that we could just update the pref as soon as we send it off to the `GutenbergPropsBuilder`. That would ensure it was done as soon as the editor was started, but not until the initial value was sent to the editor.

Thinking about this some more, my concern with this approach is that the `GutenbergPropsBuilder` gets recreated and (with "new" props) when some things change ([see here](https://github.com/wordpress-mobile/WordPress-Android/blob/14d372ba5112f47300288e2fb0fff15f00039b5c/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L979)). If that happened, then I _think_ that if we made ^that^ suggested change, then the editor could start with showing the onboarding and then when some other prop changed so a new `GutenbergPropsBuilder` was used and sent across the bridge, the "new" value would hide onboarding (since it would think the editor had already been opened). Your current android implementation actually avoids that bug nicely, but I feel like it might occur on iOS (cc: @illusaen ).

Reply:
So when I tested it I didn't see the boolean flip even after I logged out and back in to reopen the editor.

Reply:
I spoke with @illusaen further and was able to confirm that neither one of us encountered an issue of the flag changing in the middle of the session for WPiOS. So, I believe we are safe to move forward with the current implementation in both WPAndroid and WPiOS.

I have discovered that WPAndroid includes a [`onEditorDidMount` callback](https://github.com/wordpress-mobile/WordPress-Android/blob/c43e8bc1f03b4bc8bba37f1b92b85aa2830f0726/libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java#L336-L353), which is similar to WPiOS' own `gutenbergDidMount`. So, I may open a follow up PR to more closely align the implementations, but in the meantime I plan to merge this work as the refactor shouldn't impact the functionality.

Reply:
Following up to share that I no longer plan to refactor this code. We ran into [race condition issues on iOS](https://github.com/wordpress-mobile/WordPress-iOS/pull/16574/commits/5041436591d3869d4d43d4fab2cd7d01ca1877b0) where the `hasLaunchedGutenbergEditor` flag did, in fact, get set to the incorrect value during the middle of the first session. So, leveraging an end-of-life lifecycle hook seems most appropriate for this feature.